### PR TITLE
Make margins obey the snap option in the TextureRegion editor

### LINKED ISSUE #25845: Texture region margin does not obey pixel snap
Godot version: 3.1 beta 4

Texture region margin does not obey pixel snap while the region rect does.

1. create a new texture style
2. try to make a 9 slice by setting the texture region and texture margin

image

### PR DESCRIPTION
**Pull request by YeldhamDev**
Fixes #25845.

### CODE REVIEW

File: editor/plugins/texture_region_editor_plugin.cpp [hunk 1]
```diff
@@ -480,20 +480,41 @@ void TextureRegionEditor::_region_input(const Ref<InputEvent> &p_input) {
...
-					new_margin = prev_margin + (mm->get_position().y - drag_from.y) / draw_zoom;
-				} else if (edited_margin == 1) {
-					new_margin = prev_margin - (mm->get_position().y - drag_from.y) / draw_zoom;
-				} else if (edited_margin == 2) {
-					new_margin = prev_margin + (mm->get_position().x - drag_from.x) / draw_zoom;
-				} else if (edited_margin == 3) {
-					new_margin = prev_margin - (mm->get_position().x - drag_from.x) / draw_zoom;
+
+				if (snap_mode != SNAP_GRID) {
+					if (edited_margin == 0) {
```
Comment (akien-mga) [inline editor/plugins/texture_region_editor_plugin.cpp:488]: It's a bit weird that those checks are done based on ints and not enums, but that's for another PR :)

Comment (YeldhamDev) [reply editor/plugins/texture_region_editor_plugin.cpp:488]: Just following how it's done everywhere else in the file. :stuck_out_tongue:

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**